### PR TITLE
MPI : Add a block timeout control command.

### PR DESCRIPTION
**Pull request by LongChair**
When retrieving frames current API only allows to define wether the API
should block without any timeout or not block at all.
This allows to specify a block timeout via the
MPP_SET_OUTPUT_BLOCK_TIMEOUT control operation.

### CODE REVIEW

**Comment by hizukiayaka** [comment]
Could you rebase it again? I will merge it.

**Comment by LongChair** [comment]
Regarding the get_frame function, if m_frames->wait returns a timeout error (ETIMEDOUT), what error code should  the get_frame function return ? I haven't found any proper error code in mpp_err.h that would mean timeout. Should I add an MPP_ERR_TIMEOUT there ?

**Comment by LongChair** [comment]
I Updated the PR, feel free to comment again but don't merge it yet, i need to make a few sanity checks first :)

File: mpp/mpp.cpp [hunk 1]
```diff
@@ -626,6 +638,9 @@ MPP_RET Mpp::control_mpp(MpiCmd cmd, MppParam param)
         MppPollType block = *((MppPollType *)param);
         mOutputBlock = block;
     } break;
+    case MPP_SET_OUTPUT_BLOCK_TIMEOUT: {
+        mOutputBlockTimeout = *((int *)param);
```
Comment (HermanChen) [inline mpp/mpp.cpp]: The mOutputBlockTimeout is 64bit but *param will be 32bit. Although the value range is enough but it is better to get a 64bit value here?

Comment (HermanChen) [inline mpp/mpp.cpp]: If it is related to external standard usage or we can add a S32 to milisecond translation at the control?

Comment (hizukiayaka) [inline mpp/mpp.cpp]: I would suggest to use 64 bits as it is more common type for the milisecond in the other media platform.

Comment (HermanChen) [inline mpp/mpp.cpp]: BTW, the error handling can be more general. I do not think carefully about the error code design. If you have any suggestion do not be hesitated to add a commit : )

Comment (HermanChen) [inline mpp/mpp.cpp]: @LongChair Forget to @ : )

**Pull request merged**